File: level1/ex51.js
function solution(sizes) {
  let maximumWidth = 0;
  let maximumHeight = 0;

  sizes.forEach(size => {
    // 더 작은 길이를 height로 설정
    const width = size[0] > size[1] ? size[0] : size[1];
    const height = size[0] > size[1] ? size[1] : size[0];

    if (width > maximumWidth) maximumWidth = width;
    if (height > maximumHeight) maximumHeight = height;
  });

  return maximumWidth * maximumHeight;
}

File: level1/ex51.py
def solution(sizes):
    # [가로 길이, 세로 길이]를 오름차순 정렬
    sizes = list(map(lambda x: sorted(x), sizes))
    # 최대 가로 길이
    maxWidth = max(size[0] for size in sizes)
    # 최대 세로 길이
    maxHeight = max(size[1] for size in sizes)
    return maxWidth * maxHeight